### ISSUE TITLE
Android 14 compatibility - "restricted settings"

### ISSUE THREAD

**Issue by NeanderthalNinny** [cvzi/ScreenshotTile#410]
Version 2.3.5 is incompatible with Android 14, according to F-droid.

I confirmed this incompatibility after upgrading my Google Pixel 7 Pro phone today. This app is the best! I hope an updated version will be released soon to work properly with Android 14.

Thank you!
NN

**Comment by NeanderthalNinny** [comment]
It will install, but doesn't work properly.

**Comment by cvzi** [comment]
Thanks, I will look into it.

**Comment by cvzi** [comment]
I have released an update. Not sure if it fixes all problems. I can't test it thoroughly, because I am not able to run Android 14 reliably on the emulator, it keeps crashing, maybe my computer is just too slow for it 🙄

**Comment by NeanderthalNinny** [comment]
Thank you! I updated to version 2.37

Note; I had to enable "Allow restricted settings"  for the screenshot app
first. Here are details:

https://support.google.com/android/answer/12623953?sjid=10167438578691705338-NA#allowrestrictedsettings

Then I was able to enable accessibility options for screenshot (this was
grayed out before).

Now everything is working fine! ❤️

Thanks!

**Comment by NeanderthalNinny** [comment]
I would like to donate $15 toward the app. Do you accept PayPal or Zelle?

**Comment by cvzi** [comment]
Oh okay, I see the "restricted settings" thing now. It only happens when the app is installed from F-droid (and probably Google Play as well) but when I install the app as a developer on the emulator, it doesn't happen...

I guess I need to add an explanation so that people on Android 14 know what to do. Maybe the app can even open that settings screen where the "restricted settings" can be enabled.

I appreciate your offer, I have PayPal cuzi@openmail.cc or any of the sites listed here: https://cvzi.github.io/.github/

**Comment by NeanderthalNinny** [comment]
Yes, I think an explanation would be very helpful for Android 14 users.
The issue might have been the restricted settings all along and I was
unaware that lifting the restricted option was even available.

In appreciation for your excellent app and prompt support, I have sent you
15 euros with my thanks 😊

Thank you very much.
❤️

**Comment by cvzi** [comment]
Thank you!

Notes to myself:
After reading about these "restricted settings", it seems this can happen on Android 13 already. It depends on whether the device manufacturer has enabled it or not.
Some people claim it is only applied to apps that are not installed via Google Play.
There seems to be no way for the app to know whether the "restricted settings" are enabled or not.

**Comment by NeanderthalNinny** [comment]
Thanks for your explanation.

We learn something new every day 😉

By the way, Android 14 seems rather peppy than the previous version. Just
saying...

I pre-ordered the Pixel 8 Pro phone and will pick it up next week. Best Buy
had a decent trade-in deal and gives you the pixel watch free which I'm
going to sell on Swappa.

**Comment by cvzi** [comment]
Yeah I saw very similar deals here. I am kind of hesitant with these expensive phones though, cause I tend to break things a bit too quickly 🙄

---

I noticed another peculiar thing about these "restricted settings": you cannot enable them pre-emptively. On a fresh install of the app, the "restricted settings" menu is not available at all for the app.

Only after you try to enable the accessibility settings i.e. you have tapped on the grayed-out accessibility option, will the "restricted settings" menu become available.

That's a bit silly. I'll have to send users to the accessibility settings, then to the  "restricted settings" and then back to the accessibility settings...
